frontend/src/languageDetection/detect.ts

```diff
@@ -15,19 +17,34 @@ const modulOperations = new ModelOperations({
   weightsLoaderFunc: fetchWeights,
 });
 
+export type DetectionResult = StreamLanguage<unknown> | LanguageSupport | null;
+
+const langMap = {
+  ts: langs.typescript(),
```

Comment:
Do we have an elegant way of building this mapping 🤔? Do you think we should support all languages that `@uiw/codemirror-extensions-langs` have?